magpie/api/management/user/user_utils.py

```diff
@@ -853,8 +851,8 @@ def get_user_service_resources_permissions_dict(user, service, request,
 
 
 def check_user_info(user_name=None, email=None, password=None, group_name=None,  # required unless disabled explicitly
-                    check_name=True, check_email=True, check_password=True, check_group=True):
-    # type: (Str, Str, Str, Str, bool, bool, bool, bool) -> None
+                    check_name=True, check_email=True, check_password=True, check_group=True, check_anonymous=True):
```

Comment:
Not against regrouping the check under this call, but I got a vague memory that the reserved anonymous checks were done where needed in the code. Potentially duplicate code could be removed in favor of passing this argument?

Reply:
I've replaced most of the previous anonymous checks because they no longer apply.

Previously, they were just checking whether it matched the reserved anonymous username. Now it checks against the regular expression defined by the `protected_user_name_regex` function